domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/gtfs/frequencies/Frequency.java

```diff
@@ -0,0 +1,218 @@
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.domain.entity.gtfs.frequencies;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.mobilitydata.gtfsvalidator.domain.entity.gtfs.EntityBuildResult;
+import org.mobilitydata.gtfsvalidator.domain.entity.gtfs.GtfsEntity;
+import org.mobilitydata.gtfsvalidator.domain.entity.notice.base.Notice;
+import org.mobilitydata.gtfsvalidator.domain.entity.notice.error.IntegerFieldValueOutOfRangeNotice;
+import org.mobilitydata.gtfsvalidator.domain.entity.notice.error.MissingRequiredValueNotice;
+import org.mobilitydata.gtfsvalidator.domain.entity.notice.error.UnexpectedEnumValueNotice;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Class for all entities defined in transfers.txt. Can not be directly instantiated: user must use the
+ * {@link FrequencyBuilder} to create this.
+ */
+public class Frequency extends GtfsEntity {
+    @NotNull
+    private final String tripId;
+    @NotNull
+    private final Integer startTime;
+    @NotNull
+    private final Integer endTime;
+    @NotNull
+    private final Integer headwaySecs;
+    @Nullable
+    private final ExactTimes exactTimes;
+
+    private Frequency(@NotNull String tripId,
+                      @NotNull Integer startTime,
+                      @NotNull Integer endTime,
+                      @NotNull Integer headwaySecs,
+                      @Nullable ExactTimes exactTimes) {
+        this.tripId = tripId;
+        this.startTime = startTime;
+        this.endTime = endTime;
+        this.headwaySecs = headwaySecs;
+        this.exactTimes = exactTimes;
+    }
+
+    @NotNull
+    public String getTripId() {
+        return tripId;
+    }
+
+    @NotNull
+    public Integer getStartTime() {
+        return startTime;
+    }
+
+    @NotNull
+    public Integer getEndTime() {
+        return endTime;
+    }
+
+    @NotNull
+    public Integer getHeadwaySecs() {
+        return headwaySecs;
+    }
+
+    @Nullable
+    public ExactTimes isExactTimes() {
+        return exactTimes;
+    }
+
+    /**
+     * Builder class to create {@link Frequency} objects. Allows an unordered definition of the different attributes of
+     * {@link Frequency}.
+     */
+    public static class FrequencyBuilder {
+        private String tripId;
+        private Integer startTime;
+        private Integer endTime;
+        private Integer headwaySecs;
+        private ExactTimes exactTimes;
+        private String originalStartTime;
+        private String originalEndTime;
+        private Integer originalExactTimes;
+        private final List<Notice> noticeCollection = new ArrayList<>();
```

Comment:
Probably not a big deal for frequencies.txt due to the typical number of records, but you could potentially save a lot of memory by lazy-initializing this `List` for files like stop_times.txt and shapes.txt, and in some cases calendar_dates.txt. In most cases for valid data types this list will never be used, but it's being instantiated anyway.

Reply:
Per below comment, if there is only one instance of the builder being used, then this doesn't really help.